hotsos/core/plugins/storage/ceph.py

```diff
@@ -430,10 +430,10 @@ def laggy_pgs(self):
 
         laggy_pgs = []
         for pg in self.pg_dump['pg_map']['pg_stats']:
-            for state in ['laggy', 'wait']:
-                if state in pg['state']:
-                    laggy_pgs.append(pg)
-                    break
+            # Looking for 'laggy' or 'wait', but not 'backfill_wait', etc
+            r = re.findall(r'(laggy|^wait|[+]wait$|[+]wait[+])', pg['state'])
```

Comment:
-1 on this. Although this does match currently, it's not really precise in that this would still match other strings with "laggy" in them.

I think I'd prefer to switch to splitting the string on + and then check the list, e.g.
`
states = pg['state'].split('+')
any(elem in states for elem in ['wait', 'laggy'])
`

Then check if laggy and wait are in the list.
It is apparently not super syntactically simple to check for two items but:

I tried to write a regex to do this but had trouble making it non-greedy so it would correct match both wait and laggy in the same string one after another, e.g. 
`r = re.findall(r'(?:(?:^|\+)(laggy|wait)(?:\+|$)?)', 'active+remapped+laggy+wait+laggy')`

Reply:
I didn't particularly worry about 'laggy' because there are no other [states](https://docs.ceph.com/en/quincy/rados/operations/pg-states/#placement-group-states) with 'laggy' states. Even if there's one in future, it's probably a specialisation of the same and we'd probably want to include that as well.

Reply:
But in the interest of accuracy, I've used 'split' as suggested ;-)